### ISSUE TITLE
CI tests broken, failures related to GAP code

### ISSUE THREAD

**Issue by fingolfin** [oscar-system/Oscar.jl#2027]
I am seeing these warnings when loading OSCAR now (possibly since merging PR #1988 by @ThomasBreuer ?)
<s>
```
WARNING: Method definition julia_to_gap(Base.Set{T}) where {T} in module GAP at /Users/mhorn/Projekte/OSCAR/GAP.jl/src/julia_to_gap.jl:163 overwritten in module Oscar at /Users/mhorn/Projekte/OSCAR/Oscar.spielwiese/src/GAP/oscar_to_gap.jl:52.
  ** incremental compilation may be fatally broken for this module **

WARNING: Method definition julia_to_gap(Base.Set{T}, Base.IdDict{Any, Any}) where {T} in module GAP at /Users/mhorn/Projekte/OSCAR/GAP.jl/src/julia_to_gap.jl:163 overwritten in module Oscar at /Users/mhorn/Projekte/OSCAR/Oscar.spielwiese/src/GAP/oscar_to_gap.jl:52.
  ** incremental compilation may be fatally broken for this module **

WARNING: Method definition kwcall(Any, typeof(GAP.julia_to_gap), Base.Set{T}) where {T} in module GAP at /Users/mhorn/Projekte/OSCAR/GAP.jl/src/julia_to_gap.jl:163 overwritten in module Oscar at /Users/mhorn/Projekte/OSCAR/Oscar.spielwiese/src/GAP/oscar_to_gap.jl:52.
  ** incremental compilation may be fatally broken for this module **

WARNING: Method definition kwcall(Any, typeof(GAP.julia_to_gap), Base.Set{T}, Base.IdDict{Any, Any}) where {T} in module GAP at /Users/mhorn/Projekte/OSCAR/GAP.jl/src/julia_to_gap.jl:163 overwritten in module Oscar at /Users/mhorn/Projekte/OSCAR/Oscar.spielwiese/src/GAP/oscar_to_gap.jl:52.
  ** incremental compilation may be fatally broken for this module **
```
</s>

UPDATE: sorry, those warnings came from my Oscar looking at a dev version of GAP.jl. But the CI failures are real, see e.g. https://github.com/oscar-system/Oscar.jl/actions/runs/4379175069

**Comment by fingolfin** [comment]
Urgh, in fact CI is failing now everywhere since #1988 was merged. But that PR was passing all CI tests just before the merge?!?

**Comment by fingolfin** [comment]
Ah, most likely this comes from PR #2008 which was merged *after* the CI tests for PR #1988 did last run.

This incident really makes me want to try out the [Merge Queue feature](https://github.blog/changelog/2023-02-08-pull-request-merge-queue-public-beta/)

**Comment by fingolfin** [comment]
```
order 2: Error During Test at /Users/mhorn/Projekte/OSCAR/Oscar.jl/test/GAP/iso_gap_oscar.jl:13
  Test threw exception
  Expression: preimage(iso, oxi) == xi
  UndefKeywordError: keyword argument `recursive` not assigned
  Stacktrace:
    [1] julia_to_gap(obj::FqFieldElem)
      @ GAP ~/.julia/packages/GAP/EQ2to/src/julia_to_gap.jl:109
    [2] julia_to_gap(obj::FqFieldElem, recursion_dict::IdDict{Any, Any}; recursive::Bool)
      @ GAP ~/.julia/packages/GAP/EQ2to/src/julia_to_gap.jl:110
    [3] Union{Bool, GAP.FFE, GAP.GapInt}(obj::FqFieldElem; recursive::Bool)
      @ GAP ~/.julia/packages/GAP/EQ2to/src/constructors.jl:6
    [4] Union{Bool, GAP.FFE, GAP.GapInt}(obj::FqFieldElem)
      @ GAP ~/.julia/packages/GAP/EQ2to/src/constructors.jl:6
    [5] (::Oscar.var"#1484#1497"{FqFieldElem, Nemo.FinFieldMorphism{FqField, FqField}})(i::Int64)
      @ Oscar ./none:0
    [6] iterate
      @ ./generator.jl:47 [inlined]
    [7] collect(itr::Base.Generator{UnitRange{Int64}, Oscar.var"#1484#1497"{FqFieldElem, Nemo.FinFieldMorphism{FqField, FqField}}})
      @ Base ./array.jl:782
    [8] (::Oscar.var"#1483#1496"{Nemo.FinFieldMorphism{FqField, FqField}, GAP.GapObj, Int64})(x::FqFieldElem)
      @ Oscar ~/Projekte/OSCAR/Oscar.jl/src/GAP/iso_oscar_gap.jl:126
    [9] preimage(M::MapFromFunc{GAP.GapObj, FqField}, a::FqFieldElem; check::Bool)
      @ Oscar ~/Projekte/OSCAR/Oscar.jl/src/GAP/iso_gap_oscar.jl:6
   [10] preimage(M::MapFromFunc{GAP.GapObj, FqField}, a::FqFieldElem)
      @ Oscar ~/Projekte/OSCAR/Oscar.jl/src/GAP/iso_gap_oscar.jl:3
   [11] macro expansion
      @ ~/.julia/juliaup/julia-1.9.0-rc1+0.aarch64.apple.darwin14/share/julia/stdlib/v1.9/Test/src/Test.jl:478 [inlined]
   [12] macro expansion
      @ ~/Projekte/OSCAR/Oscar.jl/test/GAP/iso_gap_oscar.jl:13 [inlined]
   [13] macro expansion
      @ ~/.julia/juliaup/julia-1.9.0-rc1+0.aarch64.apple.darwin14/share/julia/stdlib/v1.9/Test/src/Test.jl:1586 [inlined]
   [14] macro expansion
      @ ~/Projekte/OSCAR/Oscar.jl/test/GAP/iso_gap_oscar.jl:2 [inlined]
   [15] macro expansion
      @ ~/.julia/juliaup/julia-1.9.0-rc1+0.aarch64.apple.darwin14/share/julia/stdlib/v1.9/Test/src/Test.jl:1498 [inlined]
   [16] top-level scope
      @ ~/Projekte/OSCAR/Oscar.jl/test/GAP/iso_gap_oscar.jl:2
order 2: Error During Test at /Users/mhorn/Projekte/OSCAR/Oscar.jl/test/GAP/iso_gap_oscar.jl:13
  Test threw exception
  Expression: preimage(iso, oxi) == xi
  UndefKeywordError: keyword argument `recursive` not assigned
  Stacktrace:
```

So it invokes `julia_to_gap` on a `FqFieldElem` and that runs into an error as we don't expect such values.

What I don't get is:
- how did this work before at all?
- why did it break now, and not before?
